spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
+  let(:user) { FactoryBot.create(:user) }
+  let(:post) { FactoryBot.create(:post) }
+
+  describe 'GET #index' do
+    it 'returns a success response' do
+      get :index, params: { user_id: user }
+      expect(response).to be_successful
+    end
+
+    it 'returns http success' do
+      get :index, params: { user_id: user.id }
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the index template' do
+      get :index, params: { user_id: user.id }
+      expect(response).to render_template(:index)
+    end
+
+    it 'assigns the correct post count' do
+      get :index, params: { user_id: user.id }
+      expect(assigns(:posts).count).to be > 1
+    end
+  end
```

Comment:
- [ ] This test is failing. Kindly fix that.

  ![image](https://user-images.githubusercontent.com/36542986/215987126-5410707d-bfc4-4c42-858d-aa7a972e8ff7.png)

Reply:
Test now passing